### ISSUE TITLE
cosine_similarity tensorflow 

### ISSUE THREAD

**Issue by Aditya-Komaravolu** [unifyai/ivy#4521]

**Comment by Daniel4078** [comment]
Just to answer your question in the TODO list issue, the cosine_similarity you want to add should be placed in here: ivy/functional/frontends/tensorflow/layers.py

**Comment by Aditya-Komaravolu** [comment]
@Daniel4078 Thank you. I still have a doubt. I can see all the loss functions in `metrics.py` and since cosine_similarity belongs to the loss function group, I have placed it there. So should I move my entire function and the test function to `layers.py` and `test_layers.py` respectively?

**Comment by Aditya-Komaravolu** [comment]
And if you don't mind, can you help me in finding out the error for the test function which is occurring in my current pull request #4567

**Comment by Daniel4078** [comment]
Well, I suspect it is an error with the entries in the TODO list issue as this function should be part of the layers module of tensorflow.keras (https://www.tensorflow.org/api_docs/python/tf/keras/layers#functions_2) but I cannot find cosine_similarity there. Maybe you should ask about this in the discord channel?

**Comment by Aditya-Komaravolu** [comment]
Yeah, maybe in the new version of TensorFlow it got pushed to Keras loss class. I tried asking in the general section for some previous questions, but I didn't get any reply :(

**Comment by Daniel4078** [comment]
You can be patient and wait for a replay in the discord but you can also give up on this and pick some other tasks that are less problematic from here (https://github.com/unifyai/ivy/issues?q=is%3Aopen+label%3A%22TensorFlow+Frontend%22+label%3AToDo) if you do not want to wait longer.

**Comment by Daniel4078** [comment]
Well, the assigning of issues and pull request is automatic and random.

**Comment by Aditya-Komaravolu** [comment]
Yeah, I noticed it. In a way, it is helping me to meet many folks in the form of PRs 😄🙌.
